logstash-core/lib/logstash/api/modules/logging.rb

```diff
@@ -0,0 +1,51 @@
+# encoding: utf-8
+#
+java_import org.apache.logging.log4j.core.LoggerContext
+
+module LogStash
+  module Api
+    module Modules
+      class Logging < ::LogStash::Api::Modules::Base
+        # retrieve logging specific parameters from the provided settings
+        #
+        # return any unused configurations
+        def handle_logging(settings)
+          Hash[settings.map do |key, level|
+            if key.start_with?("logger.")
+              _, path = key.split("logger.")
+              LogStash::Logging::Logger::configure_logging(level, path)
+              nil
+            else
+              [key, level]
+            end
+          end]
+        end
+
+        put "/" do
+          begin
+            req_body = LogStash::Json.load(request.body.read)
+            remaining = handle_logging(req_body)
+            unless remaining.empty?
+              raise ArgumentError, "unrecognized option [#{remaining.keys.first}]"
+            end
+            respond_with({"acknowledged" => true})
+          rescue Exception => e
+            status 400
+            respond_with({"error" => e.message})
+          end
```

Comment:
I do not  think we should use the exception for the control flow here, because this make all errors inherited from **Exception** return `400/bad_request` and this could consume other errors which should be mark as 500 (Application error).